### PR TITLE
Up 96

### PR DESCRIPTION
**Pull request by ConnorHibbs**
This is merging both the EconomicEvent queries, as well as the AgentRelationshipRole queries. This is due to a Git failure on my part when transitioning to EconomicEvents after AgentRelationshipRoles were "done" earlier. Too much of the functionality of AgentRelationshipRole was already in the EconomicEvent, so I decided it would be better to polish both of them and submit them together, rather than trying to remove all of the AgentRelationshipRole material from the EconomicEvent branch.

Please check them both out, and then when this gets approved both items can be moved to done

### CODE REVIEW

**Comment by fontainesw** [comment]
You have merge conflicts. Please merge dev onto your branch (or otherwise resolve the conflicts)

Edit: Please merge dev onto your branch. this will allow you to fix merge conflicts, and conflicts with the project structure and convention.

File: package.json [hunk 1]
```diff
@@ -20,5 +20,9 @@
     "redux-devtools-extension": "^2.13.2",
     "tape": "^4.8.0",
     "typings": "^2.1.1"
+  },
+  "dependencies": {
```
Comment (fontainesw) [inline package.json]: These should be dev dependencies for now. also, do you still rely on apollo-fetch ? keep it : remove it.

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
+import allAgentRelationshipRoles from "../../../ui-bindings/AgentRelationshipRole/AllAgentRelationshipRoles";
+import agentRelationshipRoleById from "../../../ui-bindings/AgentRelationshipRole/AgentRelationshipRoleById";
```
Comment (fontainesw) [inline packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: I prefaced my files with "get" we should be consistent. do you want to add get? or do you think it is better form to not include that?

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+const AgentRelationshipRole = (props) => {
+  let role = props.agentRelationshipRole;
+
```
Comment (fontainesw) [inline packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: I like this solution best. good idea

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  return(
+    <div>
+      <div>
```
Comment (fontainesw) [inline packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: Extra "Div" layer

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        <p>inverseLabel: {role.symbol}</p>
+        <p>category: {role.category}</p>
+        <br/>
+      </div>
+    </div>
+  )
+};
+
+const AgentRelationshipRolesList = allAgentRelationshipRoles( ({agentRelatonshipRoles, loading, error}) => {
+  console.log("Rendering Agent Relationship Role List", agentRelatonshipRoles);
```
Comment (fontainesw) [inline packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: Please remove this console.log

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import * as React from "react";
```
Comment (fontainesw) [inline packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx:10]: Please add a brief header to your file in javadoc style

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx [hunk 7]
```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
```
Comment (fontainesw) [inline packages/app/pages/EconomicEvent/EconomicEvent.tsx]: Please add a brief header to your file in javadoc style

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+/**
+ * Main component for the page. Contains a search box and a list of
+ * all available Economic Events
+ */
+class EconomicEvent extends React.Component {
+
+  // declare the state and what variables are used
+  state = {
+    eventId: 57
```
Comment (fontainesw) [inline packages/app/pages/EconomicEvent/EconomicEvent.tsx]: Why did you choose event 57 here

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+ * the response from the database to a component that can be rendered on the screen
+ */
+const EconomicEventList = allEconomicEvents( ({economicEvents, loading, error}) => {
+  if (loading) {
+    return <h2>Loading...</h2>
+  } else if (error) {
+    return <h2>Error!</h2>
+  }
+
+  return (
```
Comment (fontainesw) [inline packages/app/pages/EconomicEvent/EconomicEvent.tsx:84]: The indentation here looks odd. is that just me?

File: packages/app/routes.tsx [hunk 10]
```diff
@@ -42,6 +45,15 @@ const routes = (
       <Route path="my-account" component={Todo} />
       <Route path="my-skills" component={Todo} />
     </Route>
+
```
Comment (fontainesw) [inline packages/app/routes.tsx]: Please merge dev onto your branch and follow the standard I have set for the API demo routes

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+        <h1>Economic Events</h1>
+
+        <h3>Search by ID:</h3>
+        <form onSubmit={this.handleClick}>
+          <input type="text" id="idForm" defaultValue="57"/>
+        </form>
+
+        <h4>Results</h4>
+        ======================================================<br/>
+        <EconomicEventById eventId={this.state.eventId} />
```
Comment (fontainesw) [inline packages/app/pages/EconomicEvent/EconomicEvent.tsx]: While using elements before they appear is allowed by typescript, it is disallowed by our Linter. please move the declaration for this( and the other methods you use above the class to be in compliance with the linter

File: packages/ui-bindings/AgentRelationshipRole/AgentRelationshipRoleById.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
+import { connect } from "react-redux"
+import { gql, graphql, compose } from "react-apollo"
+import { AppState } from "@vflows/store/types"
+import { getActiveLoginToken } from "@vflows/store/selectors/auth"
+
+const query = gql`
+query($token: String, $id: Int){
+  viewer(token: $token) {
+    agentRelationshipRole(id: $id) {
+      id
```
Comment (fontainesw) [inline packages/ui-bindings/AgentRelationshipRole/AgentRelationshipRoleById.tsx]: Did you consider making a fragment for this to reduce code duplication?

File: packages/ui-bindings/AgentRelationshipRole/AllAgentRelationshipRoles.tsx [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import { connect } from "react-redux"
+import { gql, graphql, compose } from "react-apollo"
+import { AppState } from "@vflows/store/types"
+import { getActiveLoginToken } from "@vflows/store/selectors/auth"
+
+const query = gql`
+query($token: String){
+  viewer(token: $token) {
+    allAgentRelationshipRoles {
+      id
```
Comment (fontainesw) [inline packages/ui-bindings/AgentRelationshipRole/AllAgentRelationshipRoles.tsx]: See fragment comment above

File: packages/ui-bindings/EconomicEvent/AllEconomicEvents.tsx [hunk 14]
```diff
@@ -0,0 +1,48 @@
+import { connect } from "react-redux"
+import { gql, graphql, compose } from "react-apollo"
+import { AppState } from "@vflows/store/types"
+import { getActiveLoginToken } from "@vflows/store/selectors/auth"
+
+const query = gql`
+query($token: String) {
+  viewer(token: $token) {
+    allEconomicEvents {
+      id
```
Comment (fontainesw) [inline packages/ui-bindings/EconomicEvent/AllEconomicEvents.tsx]: This isn't all of the economic event fields. please make a fragment to query all properties with minimal code duplication

File: packages/ui-bindings/EconomicEvent/EconomicEvent.tsx.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+
+export default interface EconomicEvent {
```
Comment (fontainesw) [inline packages/ui-bindings/EconomicEvent/EconomicEvent.tsx.ts]: this is not an accurate representation of an economic event. consider also adding a fragment to this file to use in your queries

File: packages/ui-bindings/EconomicEvent/EconomicEventById.tsx [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+import { gql, graphql, compose } from "react-apollo"
+import { AppState } from "@vflows/store/types"
+import { getActiveLoginToken } from "@vflows/store/selectors/auth"
+import { create } from "domain";
+
+export const query = gql`
+query($token: String, $id: Int) {
+  viewer(token: $token) {
+    economicEvent(id: $id) {
+      id
```
Comment (fontainesw) [inline packages/ui-bindings/EconomicEvent/EconomicEventById.tsx]: See comments above on fragments and completeness

File: packages/ui-bindings/agent/agent.tsx [hunk 17]
```diff
@@ -51,12 +51,12 @@ query($token: String, $agentId: Int) {
         unplannedEconomicEvents {
           action
         }
-        processEconomicEvents {
-          action
-        }
-        processCommitments {
-          action
-        }
+        #processEconomicEvents {
```
Comment (fontainesw) [inline packages/ui-bindings/agent/agent.tsx]: Why did you comment this out?

File: packages/ui-bindings/agent/agent.tsx [hunk 18]
```diff
@@ -84,6 +84,10 @@ query($token: String, $agentId: Int) {
         image
         note
         category
+        currentLocation {
```
Comment (fontainesw) [inline packages/ui-bindings/agent/agent.tsx]: Why did you add this?

**Comment by fontainesw** [comment]
Overall, good work. I would suggest merging dev onto your branch. It is good practice to do this periodically on large projects. I would suggest following the standards set in the dev branch. I think it may also solve some other issues you have had. otherwise, good job getting some story points cranked out

File: package.json (hunk 1 above)
Comment (ConnorHibbs) [reply package.json]: Removed apollo-fetch and moved the @types/react into a dev dependency
:+1:

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx (hunk 2 above)
Comment (ConnorHibbs) [reply packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: Good naming convention! I changed my file names to match the viewer queries they were exporting
:+1:

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx (hunk 3 above)
Comment (ConnorHibbs) [reply packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: :smile: :+1:

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx (hunk 4 above)
Comment (ConnorHibbs) [reply packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: Removed
:+1:

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx (hunk 5 above)
Comment (ConnorHibbs) [reply packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx]: Removed
:+1:

File: packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx (hunk 6 above)
Comment (ConnorHibbs) [reply packages/app/pages/AgentRelationshipRole/AgentRelationshipRole.tsx:10]: Added a header, matching yours in format
:+1:

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 7 above)
Comment (ConnorHibbs) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx]: Added a header, matching yours in format
:+1:

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 8 above)
Comment (ConnorHibbs) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx]: Because it was one that displayed an event with more data attached to it. Should I remove it and set it to null or undefined?
:question:

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 9 above)
Comment (ConnorHibbs) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx:84]: I'm not sure. I re-indented it so in case it was wrong it should be fixed now.
:+1:

File: packages/app/routes.tsx (hunk 10 above)
Comment (ConnorHibbs) [reply packages/app/routes.tsx]: Merged dev into my feature branch, and formatted routes to go under api as well.
:+1:

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 11 above)
Comment (ConnorHibbs) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx]: I moved the individual components above the React.Component for the page. If this is not what you meant, please let me know

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 8 above)
Comment (fontainesw) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx]: I personally would set it to null.

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 9 above)
Comment (fontainesw) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx:84]: Sweet

File: packages/app/pages/EconomicEvent/EconomicEvent.tsx (hunk 11 above)
Comment (fontainesw) [reply packages/app/pages/EconomicEvent/EconomicEvent.tsx]: Sorry, I was really vague. our Linter disallows using react elements (i.e. "EconomicEventById") before they are declared in the file.

File: packages/ui-bindings/Agent/agent.tsx [hunk 19]
```diff
@@ -51,12 +51,12 @@ query($token: String, $agentId: Int) {
...
-        }
-        processCommitments {
-          action
-        }
+        #processEconomicEvents {
+        #  action
+        #}
+        #processCommitments {
+        #  action
+        #}
```
Comment (Ryan-Guinn) [inline packages/ui-bindings/Agent/agent.tsx:59]: Commented out code is against coding style

File: packages/ui-bindings/EconomicEvent/EconomicEvent.tsx.ts (hunk 15 above)
Comment (fontainesw) [reply packages/ui-bindings/EconomicEvent/EconomicEvent.tsx.ts]: "This is not an accurate representation" - fields are missing

File: packages/ui-bindings/AgentRelationshipRole/AgentRelationshipRoleById.tsx (hunk 12 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/AgentRelationshipRole/AgentRelationshipRoleById.tsx]: Fragment added
:+1:

File: packages/ui-bindings/AgentRelationshipRole/AllAgentRelationshipRoles.tsx (hunk 13 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/AgentRelationshipRole/AllAgentRelationshipRoles.tsx]: Fragment added
:+1:

File: packages/ui-bindings/EconomicEvent/AllEconomicEvents.tsx (hunk 14 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/EconomicEvent/AllEconomicEvents.tsx]: Fragment added, and properties displayed in the ui
:+1:

File: packages/ui-bindings/EconomicEvent/EconomicEvent.tsx.ts (hunk 15 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/EconomicEvent/EconomicEvent.tsx.ts]: Interface temporarily removed until the rest of the types are defined
:x:

File: packages/ui-bindings/EconomicEvent/EconomicEventById.tsx (hunk 16 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/EconomicEvent/EconomicEventById.tsx]: Fragment created
:+1:

File: packages/ui-bindings/agent/agent.tsx (hunk 17 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/agent/agent.tsx]: The API changed. After I pulled the latest from api-extensions upstream, the naming had changed

File: packages/ui-bindings/agent/agent.tsx (hunk 18 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/agent/agent.tsx]: The API changed. After I pulled the latest from api-extensions upstream, some new elements were added

File: packages/ui-bindings/Agent/agent.tsx (hunk 19 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/Agent/agent.tsx:59]: The naming had changed. I left the original in there for the time being so other developers who had not pulled in the lastest from api-extensions would know what might fix their code if they had errors.

**Review by Lou3797 — APPROVED** [review]

**Review by fontainesw — APPROVED** [review]
Looks good

**Review by Ryan-Guinn — APPROVED** [review]

**Comment by Lou3797** [comment]
Looks good, all commented out query fields have comments explaining why so that's good!
